semgrep/semgrep/rule_match.py

```diff
@@ -109,13 +111,20 @@ def lines(self) -> List[str]:
 
         Assumes file exists.  Note that start/end line is one-indexed
         """
-        if "lines" in self.extra:
-            return self.extra["lines"]
+        start_line = self.start["line"] - 1
+        end_line = self.end["line"]
 
-        with self.path.open(
-            buffering=1, errors="replace"
-        ) as fin:  # buffering=1 turns on line-level reads
-            return list(itertools.islice(fin, self.start["line"] - 1, self.end["line"]))
+        try:
+            return self._lines_cache[start_line][end_line]
+        except KeyError:
+            pass
+
+        # buffering=1 turns on line-level reads
+        with self.path.open(buffering=1, errors="replace") as fd:
+            result = list(itertools.islice(fd, start_line, end_line))
```

Comment:
With above suggestion, you'd either also rename `end_line` here, or add 1 back on.